atlp-rwanda/Phantom-Backend-Pirates
# Project structure setup

#### What does this PR do?
Set up the project structure to help start building RESTful web APIs using Express framework with best structure practices

#### Description of Task to be completed?
1.  Set a project structure
2.  Set up a welcome endpoint
3.  Configure ESLint for code analysis
4.  Set API documentation with Swagger

#### How should this be manually tested?
1.  After cloning the repo, cd into it and first install all the packages used by running `npm install` in the your terminal.
2.  To run code analysis run `npm run lint` in your terminal. This should analyze both all js and jsx extension files.
3.   To run the development server run `npm run devServer` in your terminal and `localhost:3000` in the browser
4.   To test  swagger documentation run `localhost:3000/api-docs`

#### Screenshots (if appropriate)
![swagger-documentation](https://user-images.githubusercontent.com/16195731/154359088-0c24d9a4-46f4-426e-aacf-d95981bb990a.png)
![swagger-documentation-with the entry API](https://user-images.githubusercontent.com/16195731/154359108-a65f21fd-dc48-43f8-ad7a-c4d98cb32c72.png)
![Welcome Endpoint](https://user-images.githubusercontent.com/16195731/154359069-eb9ccd64-fdc2-46e4-aba6-04380f0b4785.png)
